starter-code/src/viking.js

```diff
@@ -1,11 +1,78 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(theDamage) {
+    this.health = this.health - theDamage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(theDamage) {
+    this.health = this.health - theDamage;
+    if (this.health > 0){
+      return `${this.name} has received ${theDamage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return `Odin Owns You All!`;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(theDamage) {
+    this.health = this.health - theDamage;
+    if (this.health > 0){
+      return `A Saxon has received ${theDamage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    var chosenSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+    var chosenViking = Math.floor(Math.random()*this.vikingArmy.length);
+    this.saxonArmy[chosenSaxon].receiveDamage(this.vikingArmy[chosenViking].strength);
+    
+  }
+  saxonAttack() {
+    var chosenSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+    var chosenViking = Math.floor(Math.random()*this.vikingArmy.length);
+    this.vikingArmy[chosenViking].receiveDamage(this.saxonArmy[chosenSaxon].strength);
+  }
+  showStatus() {
+    if (this.saxonArmy.length === 0) {
+      return "Vikings have won the war of the century!";
+    } else if (this.vikingArmy.length === 0) {
```

Comment:
La valeur peut aussi descendre en dessous de zéro, on peut donc écrire: ` if (this.saxonArmy.length <= 0)` et `else if (this.vikingArmy.length <= 0)`